.github/workflows/ci.yml

```diff
@@ -0,0 +1,137 @@
+      - name: build_test
+        shell: bash
+        run: ./scripts/build_test.sh
+        env:
+          TIMEOUT: ${{ env.TIMEOUT }}
+  Fuzz:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.20.12'
+          check-latest: true
+      - name: fuzz_test
+        shell: bash
+        run: ./scripts/build_fuzz.sh 15 # Run each fuzz test 15 seconds
+  e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.20.12'
+          check-latest: true
+      - name: Build AvalancheGo Binary
+        shell: bash
+        run: ./scripts/build.sh -r
+      - name: Run e2e tests
+        shell: bash
+        run: E2E_SERIAL=1 ./scripts/tests.e2e.sh
+      - name: Upload tmpnet network dir
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: tmpnet-data
+          path: ~/.tmpnet/networks/1000
+  e2e_existing_network:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.20.12'
+          check-latest: true
+      - name: Build AvalancheGo Binary
+        shell: bash
+        run: ./scripts/build.sh -r
+      - name: Run e2e tests with existing network
+        shell: bash
+        run: E2E_SERIAL=1 ./scripts/tests.e2e.existing.sh
+      - name: Upload tmpnet network dir
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: tmpnet-data
+          path: ~/.tmpnet/networks/1000
+  Upgrade:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.20.12'
+          check-latest: true
+      - name: Build AvalancheGo Binary
+        shell: bash
+        run: ./scripts/build.sh
+      - name: Run e2e tests
+        shell: bash
+        # 1.10.7 is the first version compatible with the ephnet fixture by
+        # virtue of writing a process context file on node start.
+        run: ./scripts/tests.upgrade.sh 1.10.7
+      - name: Upload ephnet network dir
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: ephnet-data
+          path: ~/.ephnet/networks/1000
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.20.12'
+          check-latest: true
+      - name: Run static analysis tests
+        shell: bash
+        run: scripts/lint.sh
+  buf-lint:
+    name: Protobuf Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bufbuild/buf-setup-action@v1.26.1
+        with:
+          github_token: ${{ github.token }}
+      - uses: bufbuild/buf-lint-action@v1
+        with:
+          input: "proto"
```

Comment:
nit: fix this missing newline at EOF ([ref](https://stackoverflow.com/a/37684763). You can usually configure your editor to add this automatically